File: linux/platform_specific.c
#include "platform_specific.h"
#include "global.h"
#include <stdio.h>
#include <stdarg.h>
#include "stb_image_write.h"
#include "image_functions.h"
#include <stdlib.h>


void disco_log (const char *fmt, ...) {
	va_list		arg;

	va_start(arg, fmt);
	vprintf(fmt, arg);
	printf("\n");
	va_end(arg);
}

// print to log with no new line at the end
void disco_log_nol (const char *fmt, ...) {
	va_list		arg;

	va_start(arg, fmt);
	vprintf(fmt, arg);
	va_end(arg);
}

void display_image (uint8_t *data, uint16_t height, uint16_t width) {

	static int		filecount = 0;		// makes each filename unique
	char			filename[100];
	int				err = 0;

	// create the filename
	sprintf(filename, "outimage%d.png", filecount++);

	err = stbi_write_png(filename, width, height, 1, data, width);

	free(data);

	if (!err) {
		printf("error writing png output.\n");
		exit(1);
	}

}

void trigger_error () {
	int a = 8;
	int b = 0;
	int c;

	c = a/b;
}

void write_file (uint8_t* data, uint32_t length, char* filename) {
	FILE *output;
	
	output = fopen(filename, "wb");
	fwrite(data, 1, length, output);
	
	fclose(output);
}

	

File: iPhoneFiles/Discotective/platform_specific.m
//
//  platform_specific.m
//  Discotective
//
//  Created by Mike Hand on 6/17/11.
//  Copyright 2011 __MyCompanyName__. All rights reserved.
//

#import "platform_specific.h"
#import "ImageHandler.h"
#import <unistd.h>

#include <stdlib.h>
#include <stdarg.h>
#import <dispatch/dispatch.h>
#import<AVFoundation/AVFoundation.h>

#define	audioFilePath @"audio.wave"


void disco_log (const char *fmt, ...){
	va_list		arg;
	
	va_start(arg, fmt);
	vprintf(fmt, arg);
	printf("\n");
	va_end(arg);
}

void disco_log_nol (const char *fmt, ...){
	va_list		arg;
	
	va_start(arg, fmt);
	vprintf(fmt, arg);
	va_end(arg);
}

void display_image (uint8_t *data, uint16_t height, uint16_t width){
	dispatch_queue_t	mainQueue;
	mainQueue		= dispatch_get_main_queue();
	UIImage* binaryUIImage	= [[ImageHandler getImageHandler] createUIImageFromBinaryImage:data withWidth:width andHeight:height];
	[[[ImageHandler getImageHandler] getImageView].image release];
	
	//Finished Preprocessing..........................
	
	dispatch_sync(mainQueue, ^{
		[[ImageHandler getImageHandler] getImageView].image	= [binaryUIImage retain];
	});
	
	//free(data);
}

void trigger_error (){
	int a=8;
	int b=0;
	int c;
	c=a/b;
}


void createAudioFile(uint8_t* data, uint32_t length){
	NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
	NSString *documentsDirectory = [paths objectAtIndex:0];
	NSString *filePath=[documentsDirectory stringByAppendingPathComponent:audioFilePath];
	NSData* dataForFile=[NSData dataWithBytesNoCopy:data length:length];
	[dataForFile writeToFile:filePath atomically:YES];
}

void playAudioFile(const char* fileName){
	NSArray *paths = NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES);
	NSString *documentsDirectory = [paths objectAtIndex:0];
	NSString* file=[NSString stringWithCString:fileName encoding:NSASCIIStringEncoding];
	NSLog(@"%@",file);
	NSString *filePath=[documentsDirectory stringByAppendingPathComponent:file];
	NSLog(@"%@", filePath);
	NSURL* fileURL=[[NSURL alloc]initFileURLWithPath:filePath];
	//TODO this doesn't appear to work
	NSLog(@"%@",fileURL);
	NSError** err=nil;
	AVAudioPlayer* audioPlayer=[[AVAudioPlayer alloc] initWithContentsOfURL:fileURL error:err];
	NSLog(@"%@",err);
	if(!err){
		NSLog(@"PLAYING");
		[audioPlayer play];
	}
}





